audio/midi_drivers/XMidiFile.cpp

```diff
@@ -723,9 +723,9 @@ void XMidiFile::ApplyFirstState(first_state &fs, int chan_mask)
 		patch->time = 0;
 		bank->time = 0;
 
-		if (do_reverb && do_chorus) reverb->next = chorus;
-		else if (do_reverb) reverb->next = bank;
-		if (do_chorus) chorus->next = bank;
+		if (reverb && do_chorus) reverb->next = chorus;
+		else if (reverb) reverb->next = bank;
+		if (chorus) chorus->next = bank;
```

Comment:
In my opinion, this change is pointless, as the pointers can never be null if the booleans are true. But in any case, you might want to also replace do_chorus in line 726 by chorus for consistency.